src/SecretManager/Microsoft.DncEng.SecretManager/Program.cs

```diff
@@ -4,22 +4,50 @@
 using System.Security.Cryptography.X509Certificates;
 using System.Threading.Tasks;
 using Microsoft.DncEng.CommandLineLib;
+using Microsoft.DncEng.SecretManager.Commands;
 using Microsoft.DncEng.SecretManager.ServiceConnections;
-using Microsoft.Extensions.Azure;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Http;
 
 namespace Microsoft.DncEng.SecretManager;
 
 public class Program : DependencyInjectedConsoleApp
 {
+    /// <summary>
+    /// Object stores global command setting as parsed from the command line at the main method
+    /// We mark this valud as protected so it can be accessed by processes that invoke the assembly outside of the command line
+    /// </summary>
+    protected static GlobalCommand _globalCommand = new GlobalCommand();
+
+
+    /// <summary>
+    /// The service tree id of calling service parsed from the command line at the main method
+    /// We mark this valeue as protected so it can be accessed by processes that invoke the assembly outside of the command line
+    /// </summary>
+    protected static Guid ServiceTreeId = Guid.Empty;
+
     public static Task<int> Main(string[] args)
     {
+        // The GlobalCommand must be pre-parsed before passing to the  ProjectBaseCommand object or the base global settings will be lost
+        var options = _globalCommand.GetOptions();
+        options.Parse(args);
+
+        // We then parse the ProjectBaseCommand to ensure we collect the service tree id at the start of the progress
+        // so it can be used for dependency ingjection
+        // The global option setings are passed to all other command objects that inhearit from the ProjectBaseCommand
+        var projectBaselCommand = new ProjectBaseCommand(_globalCommand);
```

Comment:
```suggestion
        var projectBaseCommand = new ProjectBaseCommand(_globalCommand);
```
and update references to the variable